iot-hub/Samples/device/DeviceReconnectionSample/ColorConsoleLoggerConfiguration.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    /// <summary>
+    /// A color console logger configuration that sets the default log level to Information and logs all events.
+    /// </summary>
+    public class ColorConsoleLoggerConfiguration
+    {
+        // If the EventId is set to 0, the logger will log all events.
+        private const int DefaultEventId = 0;
+
+        private static readonly IDictionary<LogLevel, ConsoleColor> s_defaultColorMapping = new Dictionary<LogLevel, ConsoleColor>
+        {
+            { LogLevel.Trace, ConsoleColor.Blue },
+            { LogLevel.Debug, ConsoleColor.DarkYellow },
+            { LogLevel.Information, ConsoleColor.Cyan },
+            { LogLevel.Warning, ConsoleColor.DarkMagenta },
+            { LogLevel.Error, ConsoleColor.Red },
+            { LogLevel.Critical, ConsoleColor.DarkRed }
+        };
+
+        public ColorConsoleLoggerConfiguration()
+        {
+            LogLevelToColorMapping = s_defaultColorMapping;
+        }
+
+        public ColorConsoleLoggerConfiguration(IDictionary<LogLevel, ConsoleColor> customConsoleColorMapping)
+            : this ()
```

Comment:
Why call the default ctor? It sets a color mapping you don't want.

Reply:
it sets the default color mapping, and then updates the dic based on what you've passed in-> as in, I might want to override only 1 color mapping, but still retain the default values for others

Reply:
got it.